exchangelib/services.py

```diff
@@ -283,9 +282,9 @@ def _paged_call(self, **kwargs):
                     break
             if not next_offset:
                 break
-            if next_offset != 1 + last_offset:
+            if next_offset != item_count:
                 # Guard against endless loop
-                log.warning('Unexpected next offset: %s -> %s', last_offset, next_offset)
+                log.warning('Unexpected next offset: %s -> %s', item_count, next_offset)
                 break
```

Comment:
I think we should promote this warning to an exception, e.g. `TransportError`, since we might be losing data unnoticed.